actions/soak_helpers.go

```diff
@@ -28,47 +36,75 @@ func BuildGoTests(executablePath, testsPath string) (string, error) {
 		Str("Command", compileCmd.String()).
 		Msg("Ran command")
 	if err != nil {
-		return "", fmt.Errorf("Env: %s\nCommand: %s\nCommand Output: %s, %w", compileCmd.Env, compileCmd.String(), string(compileOut), err)
+		return "", "", fmt.Errorf("Env: %s\nCommand: %s\nCommand Output: %s, %w", compileCmd.Env, compileCmd.String(), string(compileOut), err)
 	}
 
-	_, err = os.Stat(exePath)
+	fileInfo, err := os.Stat(exePath)
 	if err != nil {
-		return "", fmt.Errorf("Expected '%s' to exist, %w", exePath, err)
+		return "", "", fmt.Errorf("expected '%s' to exist, %w", exePath, err)
 	}
-	return exePath, nil
+	return exePath, strconv.Itoa(int(fileInfo.Size())), nil
 }
 
-// runs a soak test based on the tag, launching as many chainlink nodes as necessary
+// RunSoakTest runs a soak test based on the tag, launching as many chainlink nodes as necessary
 func RunSoakTest(testTag, namespacePrefix string, chainlinkReplicas int) error {
 	LoadConfigs()
-	soakTestsPath := filepath.Join(utils.SoakRoot, "tests")
-	exePath, err := BuildGoTests(utils.ProjectRoot, soakTestsPath)
+	_, fileSize, err := BuildGoTests(filepath.Join(utils.SoakRoot, "tests"))
 	if err != nil {
 		return err
 	}
-
-	runnerHelmValues := environment.CommonRemoteRunnerValues(
-		testTag, // Name of the test to run
-		config.ProjectConfig.RemoteRunnerConfig.SlackAPIKey,  // API key to use to upload artifacts to slack
-		config.ProjectConfig.RemoteRunnerConfig.SlackChannel, // Slack Channel to upload test artifacts to
-		config.ProjectConfig.RemoteRunnerConfig.SlackUserID,  // Slack user to notify on completion
-	)
-	env, err := environment.DeployRemoteRunnerEnvironment(
-		environment.NewChainlinkConfig(
-			environment.ChainlinkReplicas(chainlinkReplicas, config.ChainlinkVals()),
-			namespacePrefix,
-			config.GethNetworks()...,
-		),
-		filepath.Join(utils.SuiteRoot, "framework.yaml"), // Path of the framework config
-		filepath.Join(utils.SuiteRoot, "networks.yaml"),  // Path to the networks config
-		exePath, // Path to the executable test file
-		runnerHelmValues,
-	)
+	env := environment.New(&environment.Config{
+		Labels:    []string{fmt.Sprintf("envType=%s", pkg.EnvTypeEVM5RemoteRunner)},
+		Namespace: namespacePrefix,
+	})
+	err = env.
+		AddHelm(mockservercfg.New(nil)).
+		AddHelm(mockserver.New(nil)).
+		AddHelm(remotetestrunner.New(map[string]interface{}{
+			"remote_test_runner": map[string]interface{}{
```

Comment:
See my concerns with this chart style not being intuitive. This invites a whole lot of user errors inputting the wrong keys and values. Maybe best replaced with a struct.

Reply:
Nothing I can do about it now if we add a struct it needs to be transformed into a map, so another layer of abstraction, we don't need, idea is that we do not enforce structure on an arbitrary charts that may be external